lib/module.js

```diff
@@ -1,779 +1,3 @@
-  }
-  if (depth === 0) stat.cache = null;
-  return result;
-};
-
-
-// Native extension for .js
-Module._extensions['.js'] = function(module, filename) {
-  var content = fs.readFileSync(filename, 'utf8');
-  module._compile(internalModule.stripBOM(content), filename);
-};
-
-
-// Native extension for .json
-Module._extensions['.json'] = function(module, filename) {
-  var content = fs.readFileSync(filename, 'utf8');
-  try {
-    module.exports = JSON.parse(internalModule.stripBOM(content));
-  } catch (err) {
-    err.message = filename + ': ' + err.message;
-    throw err;
-  }
-};
-
-
-//Native extension for .node
-Module._extensions['.node'] = function(module, filename) {
-  return process.dlopen(module, path.toNamespacedPath(filename));
-};
-
-if (experimentalModules) {
-  Module._extensions['.mjs'] = function(module, filename) {
-    throw new ERR_REQUIRE_ESM(filename);
-  };
-}
-
-// bootstrap main module.
-Module.runMain = function() {
-  // Load the main module--the command line argument.
-  Module._load(process.argv[1], null, true);
-  // Handle any nextTicks added in the first tick of the program
-  process._tickCallback();
-};
-
-Module._initPaths = function() {
-  const isWindows = process.platform === 'win32';
-
-  var homeDir;
-  var nodePath;
-  if (isWindows) {
-    homeDir = process.env.USERPROFILE;
-    nodePath = process.env.NODE_PATH;
-  } else {
-    homeDir = safeGetenv('HOME');
-    nodePath = safeGetenv('NODE_PATH');
-  }
-
-  // $PREFIX/lib/node, where $PREFIX is the root of the Node.js installation.
-  var prefixDir;
-  // process.execPath is $PREFIX/bin/node except on Windows where it is
-  // $PREFIX\node.exe.
-  if (isWindows) {
-    prefixDir = path.resolve(process.execPath, '..');
-  } else {
-    prefixDir = path.resolve(process.execPath, '..', '..');
-  }
-  var paths = [path.resolve(prefixDir, 'lib', 'node')];
-
-  if (homeDir) {
-    paths.unshift(path.resolve(homeDir, '.node_libraries'));
-    paths.unshift(path.resolve(homeDir, '.node_modules'));
-  }
-
-  if (nodePath) {
-    paths = nodePath.split(path.delimiter).filter(function(path) {
-      return !!path;
-    }).concat(paths);
-  }
-
-  modulePaths = paths;
-
-  // clone as a shallow copy, for introspection.
-  Module.globalPaths = modulePaths.slice(0);
-};
-
-Module._preloadModules = function(requests) {
-  if (!Array.isArray(requests))
-    return;
-
-  // Preloaded modules have a dummy parent module which is deemed to exist
-  // in the current working directory. This seeds the search path for
-  // preloaded modules.
-  var parent = new Module('internal/preload', null);
-  try {
-    parent.paths = Module._nodeModulePaths(process.cwd());
-  } catch (e) {
-    if (e.code !== 'ENOENT') {
-      throw e;
-    }
-  }
-  for (var n = 0; n < requests.length; n++)
-    parent.require(requests[n]);
-};
-
-Module._initPaths();
-
-// backwards compatibility
-Module.Module = Module;
+module.exports = require('internal/modules/cjsm');
```

Comment:
I don’t think git recognizes this as a file move, so it won’t say anything about merge conflicts? That might be okay, we don’t have many PRs against this file open/to backport, I think…

Reply:
@addaleax If we don't require that each commit must pass the test, we can delete the file first (so it will be recognized as a move) and then add the redirection later.